### DISCUSSION TITLE
lxqt-config-monitor disabling and other issues

### DISCUSSION THREAD

**Discussion by bbigam** [lxqt/lxqt#2582] [General]
I have some issues with and questions about lxqt-config-monitor. Some of it will probably need to end up in bug reports, but I wanted to get it all mentioned here, to get some direction.

### Disabling lxqt-config-monitor
Is there a supported way to disable l-c-monitor? I haven't been able to find one in the GUI, config files, or Github issues. I can't figure out any way to do it other than modifying the permissions of the binary, renaming it, or deleting it. I'd rather not resort to these hacky methods.

Why disable it?

- It and kscreen always crash for me on login or wake from sleep. (More on this below)
- I don't need it, because all I use is my laptop monitor.
- Wayland non-Kwin users apparently can't use it anyway. This isn't me (I'm on Openbox and so X11), but from this, where it was set to disabled and with a warning message:  issue: https://github.com/lxqt/lxqt-config/issues/916
- Other lxqt components can be disabled individually from session settings

I would imagine that plenty of lxqt users don't need or can't use lxqt-config-monitor, and there was an issue that proposed adding the option. But it seems to have stalled out years ago: https://github.com/lxqt/lxqt/issues/1788

### Removing lxqt-config-monitor?
I'd like to suggest the devs consider splitting out l-c-monitor into a separate package, with lxqt-config recommending but not requiring it. This would allow users who don't need or can't use it to remove it, and remove the kscreen dependency it pulls in. I think this is less important than the option to disable it, though.

### Crashes
On my system (openSUSE Tumbleweed, lxqt-config 2.0.0 within a complete lxqt 2.0.x setup, libkf6screen8 6.0.5, qt 6.7.1, Openbox), l-c-monitor and kscreen always crash and coredump. This happens at login, at system wake from sleep, and if I open l-c-monitor. l-c-monitor will open, but the contents of the window are blank as if there is no screen detected. In this case the segfault and coredump happen when I close the window.

There was formerly this similar report, which was closed not due to being specifically solved, but being "probably outdated": https://github.com/lxqt/lxqt-config/issues/504

I seem to remember l-c-monitor 1.4.x being broken in Tumbleweed previously, too, but I never looked into it.

So, I'm not sure if this bug report should be in lxqt or downstream (Tumbleweed).

Thanks for your good work in lxqt.

**Comment by Chiitoo** [comment]
It can be disabled at build-time, but I suppose for openSUSE that would mean there would need to be an additional package for 'lxqt-config', and I'm not sure how packaging there works.

I remember some part(s) of the config crashing on openSUSE on a VM in the past too, but I can't remember the version, and whatever it was that I learned when looking into it...

**Reply by Chiitoo** [reply]
Oh and if there are consistent crashes, they indeed should be reported.

If using the latest release, probably to LXQt, but would also be good to report to openSUSE, especially since I've not seen such crashes on Gentoo Linux, so there might be some specific combination of things causing it.

**Reply by bbigam** [reply]
Got it, I'll work on crash reports.
I'd definitely like to request, in any case, that a user-facing UI option be added to disable l-c-monitor.

**Comment by tsujan** [comment]
> Wayland non-Kwin users apparently can't use it anyway.

This is what's shown under Wayland compositors other than `kwin_wayland` in git (the screenshot is taken under LabWC):

![wayland](https://github.com/lxqt/lxqt/assets/981076/3875b924-0579-4fd5-87dd-5d58ae49c77e)

But since 99% of LXQt users use it under X11, there's no reason to remove it because of that.

Users can always hide menu items by adding `NoDisplay=true` to desktop entries and copying them into `~/.local/share/applications`.

For investigating the crash you see, a backtrace is needed:

```
coredumpctl gdb APP_NAME
where
```

**Reply by bbigam** [reply]
Okay, I get not wanting to remove it. But, plenty of people might simply disable it, if the option is offered, as it is for various other LXQt components. I think that most users with a single monitor don't need it, is that correct?

Working on the bug report to lxqt-config for the crash.

**Reply by bbigam** [reply]
I do have it hidden, but I'd like to disable it from autostart, since I don't need it.

**Comment by bbigam** [comment]
For the crash: https://github.com/lxqt/lxqt-config/issues/1004

**Comment by sfalken** [comment]
This absolutely needs to get filed against openSUSE Tumbleweed, even if the "fix" is something that ultimately comes from upstream. 
I don't know how many of the other openSUSE maintainers watch this bugtracker, but this is the first I've heard of the issue.

And we've got a grand total of one open bug against LXQt in Tumbleweed at present

**Reply by bbigam** [reply]
And I'm the one who filed the other one!
I'll do it.

**Reply by tsujan** [reply]
> ... even if the "fix" is something that ultimately comes from upstream.

I think the nullity check in https://github.com/lxqt/lxqt-config/pull/1005 makes sense. Waiting for a user to confirm that it works...

> but this is the first I've heard of the issue.

It has never happened to me or other LXQt members either, but there's definitely a problem somewhere (in `kscreen`?).

**Reply by bbigam** [reply]
https://bugzilla.opensuse.org/show_bug.cgi?id=1226392
